t/exotic_names.t

```diff
@@ -59,7 +59,7 @@ sub caller3_ok {
 
 use Sub::Name 'subname';
 
-my @ordinal = ( 1 .. 255 );
+my @ordinal = ( 1 .. 38, 40 .. 255 );
```

Comment:
Commenting so things do not fall through the cracks: @Leont this commit needs to be reverted, and the XS-guts amended to pass as per the conversation on March 16th:
```
[01:44:11] <leont> ribasushi, aristotle: your opinions on apostrophe's in Sub::Name wasn't entirely clear to me
[01:45:24] <leont> Sub::Name always treated them as separators, but the tests suggest it should be treated as a normal character.
[01:46:13] <leont> Given how much perl's stash interfaces all resist this, I don't think this is sensible
[01:56:21] <aristotle> you can invoke a function whose name is not a valid identifier by symbolic reference
[01:56:44] <aristotle> but you cannot do that for a function whose name contains a literal :: or '
[01:57:09] <aristotle> because Perl always takes those to be package separators when it looks up the symref
[01:57:39] <aristotle> consequently I think Sub::Name should consider them package separators also
[01:57:54] <aristotle> does that make sense?
[01:57:58] <leont> Yes
```